src/module/entities/helper.js

```diff
@@ -180,6 +182,96 @@ module.exports = class UserProjectsHelper {
 		})
 	}
 
+	/**
+	 * Fetches targeted roles based on the provided entity IDs.
+	 * @param {Array<string>} entityId - An array of entity IDs to filter roles.
+	 * @returns {Promise<Object>} A promise that resolves to the response containing the fetched roles or an error object.
+	 */
+	static targetedRoles(entityId) {
+		return new Promise(async (resolve, reject) => {
+			try {
+				// Construct the filter to retrieve entities based on provided entity IDs
+				const filter = {
+					_id: {
+						$in: entityId,
+					},
+				}
+				const projectionFields = ['childHierarchyPath', 'entityType']
+				// Retrieve entityDetails based on provided entity IDs
+				const entityDetails = await entitiesQueries.entityDocuments(filter, projectionFields)
+				if (
+					!entityDetails ||
+					!entityDetails[0].childHierarchyPath ||
+					entityDetails[0].childHierarchyPath.length < 0
+				) {
+					throw {
+						status: HTTP_STATUS_CODE.not_found.status,
+						message: CONSTANTS.apiResponses.ENTITYTYPE_NOT_FOUND,
+					}
+				}
+				// Extract the childHierarchyPath and entityType
+				const { childHierarchyPath, entityType } = entityDetails[0]
+
+				// Append entityType to childHierarchyPath array
+				const updatedChildHierarchyPaths = [...childHierarchyPath, entityType]
+
+				// Construct the filter to retrieve entity type IDs based on child hierarchy paths
+				const entityTypeFilter = {
+					name: {
+						$in: updatedChildHierarchyPaths,
+					},
+					isDeleted: false,
+				}
+				const entityTypeProjection = ['_id']
+				// Retrieve entity type IDs based on child hierarchy paths
+				const fetchEntityTypeId = await entityTypeQueries.entityTypesDocument(
+					entityTypeFilter,
+					entityTypeProjection
+				)
+				// Check if entity type IDs are retrieved successfully
+				if (fetchEntityTypeId.length < 0) {
+					throw {
+						status: HTTP_STATUS_CODE.not_found.status,
+						message: CONSTANTS.apiResponses.ENTITY_TYPE_DETAILS_NOT_FOUND,
+					}
+				}
+				// Extract the _id fields from the fetched entity types to use as a filter for user roles
+				const userRoleFilter = fetchEntityTypeId.map((entityType) => entityType._id)
+
+				// Construct the filter for finding user roles based on entityTypeIds and status
+				const userRoleExtensionFilter = {
+					'entityTypes.entityTypeId': {
+						$in: userRoleFilter,
+					},
+					status: CONSTANTS.common.ACTIVE_STATUS,
+				}
+				// Specify the fields to include in the result set
+				const userRoleExtensionProjection = ['_id', 'title', 'userRoleId', 'userType']
+
+				// Fetch the user roles based on the filter and projection
+				const fetchUserRoles = await userServiceHelper.find(
```

Comment:
@MallanagoudaB Why do we have to go to user service here, we have everything in mentoring only.